### DISCUSSION TITLE
Steps in recipes

### DISCUSSION THREAD

**Discussion by Lycolus** [TandoorRecipes/recipes#1763] [General]
Hey everyone! I only recently discovered this project and I must say, I am really impressed! Great work!
I have been looking at both Tandoor and mealie as potential recipe managers and making a choice is proving difficult, each having a few advantages over the other.
Background: I have been running the 1.0B (beta) version of mealie and the development version of tandoor in this comparison.

One of the things I really like about mealie is their 'step' layout. I can use the bulk importer and type each step in the recipe process on a new line, afterwards each line is converted into a step in the recipe. For myself, being quite new to cooking, having a step-by-step guide is quite helpful and I can easily close the steps after performing them. Trying to get a similar result in tandoor has proven somewhat painful (I tend to lose overview when all the text is in a single step), but perhaps I have just been missing something so perhaps you great folks can help me out!

I have noticed I can also make steps in tandoor, but ingredients need to be coupled to a step as far as I am aware. Is it not possible to have a global recipe list for the entire recipe without having to define ingredients for each step (takes quite a bit more time)?
I think I read somewhere that these steps are more intended as a higher-level breakdown of a recipe as opposed to the steps in mealie. In that context, I can understand the way it is set up currently but I would like to verify that is the case.
In addition, I noticed I can copy-paste a list of ingredients but it does not appear that there is a similar function for steps. This means I need to press the 'add step' button every time, which, while not a great inconvenience, is quite a bit slower than the bulk import feature I talked about earlier.

Hopefully, some of the people here can help me with this workflow or otherwise let me know if this is something that might be considered for future versions of tandoor.

Thanks again for the great work on this project!

**Comment by vabene1111** [comment]
hi, thanks for your feedback, this really helps me understand how people use the app.

I have several comments in regard to your ideas and would love to hear what you think about them

1. the new version of tandoor (released yesterday) has a new importer where you can easily split a recipe into steps (basically at each linebreak). This importer can also (on desktop screen sizes) very quickly sort ingredients across steps.
2. it is planned to add copy/paste functions to ingredients and steps
3. it is planned to add something to quickly sort ingredients to steps (similarly to the sort step modal that already exists in the recipe editor).

generally i use steps for "parts" of a recipe e.g. cutting vegetables which might involve cutting cucumbers, potatoes and whatever else, or something like "baking" which might also include how to serve or finalize a dish but you can certainly use steps as very tiny atomic steps like "cut potatoes" and "cut cucumbers" depending on what you prefer (i think the latter needs the aforementioned quality of life features to be a little nicer)

**Comment by Lycolus** [comment]
Thanks for the super quick reply! Glad I am able to help the project in my own humble way.

> the new version of tandoor (released yesterday) has a new importer where you can easily split a recipe into steps (basically at each linebreak). This importer can also (on desktop screen sizes) very quickly sort ingredients across steps.

I did in fact test that version and I really liked the splitter. I had not noticed that you can move around ingredients! That's certainly an improvement to the workflow.

> it is planned to add copy/paste functions to ingredients and steps

Looking forward to it then! If you could include the possibility to split each newline into a separate step that would really be awesome.

>  it is planned to add something to quickly sort ingredients to steps (similarly to the sort step modal that already exists in the recipe editor).

That would certainly be a big improvement. It would still kinda leave me with one (small) gripe though, see below (quickly pulled this from a website, hello fellow dutch people!):
![chrome_2022-05-02_17-16-40](https://user-images.githubusercontent.com/56881763/166259466-3561add2-eb25-4845-9dab-9f27a13a5774.png)

In the use case as you describe it, it makes sense to list the ingredients which would need to be included in each step. When using these very small atomic steps, however, I feel the ingredients just clutter the available space. 
Perhaps this could be as simple as an option to hide ingredients in steps and only show the total at the top? That way I also do not need to move around the ingredients to different steps and can just leave all of them in the first step (which is what is currently the default after importing). I am not too well versed in UI-related programming so perhaps that's actually more work than it might sound. Eager to here your thoughts!

**Reply by vabene1111** [reply]
this is also something that is already planned to make it possible that you can switch the overview on and off, but that is also not done yet.

**Reply by Lycolus** [reply]
I'll keep watching the project with interest then!

**Comment by tmllull** [comment]
Hi. I discovered Tandoor a few days ago, and I think it's a very good place to save all my recipes (I used Grocy for months, but it's a little big for me xD). I'm checking for the steps in recipes, and I've a question:

EDIT: I'm running a self-hosted instance.

I created 2 steps, with specific name and ingredients. After save, the resume of the recipe shows in “total” ingredients section the title of the first step, and the first step show the name 2 times (one for the title and another like a “subtitle”).

Edit mode first step:
![imagen](https://user-images.githubusercontent.com/5560652/166987499-40012496-7df3-48c8-a8d5-7d360970ab06.png)

Edit mode second step:
![imagen](https://user-images.githubusercontent.com/5560652/166987543-2d0a174e-2458-4824-bf27-9f5238bb4ce4.png)

Resume:
![imagen](https://user-images.githubusercontent.com/5560652/166987664-3d8bc390-50a4-40f8-b3b2-db63054e5713.png)

I'm doing something wrong?

Thanks for your help, and congratulations for your work!

**Reply by vabene1111** [reply]
thanks for the kind words. This feature is called step headers and can be turned on and off but you are right, the header should not show in the step ingredient list, i need to fix that.
